checked_yaml/lib/checked_yaml.dart

```diff
@@ -75,12 +75,18 @@ ParsedYamlException toParsedYamlException(
       innerError: exception,
     );
   } else {
-    final yamlValue = yamlMap.nodes[exception.key];
-
-    if (yamlValue == null) {
-      // TODO: test this case!
+    if (exception.key == null) {
+      return ParsedYamlException(
+        exception.message ?? 'There was an error parsing the map.',
+        yamlMap,
+        innerError: exception,
+      );
+    } else if (!yamlMap.containsKey(exception.key)) {
       return ParsedYamlException(
-        exception.message!,
+        [
+          'Missing key "${exception.key}".',
+          if (exception.message != null) exception.message!,
+        ].join(' '),
```

Comment:
```suggestion
        'Missing key "${exception.key}".${exception.message ?? ''}',
```

Reply:
...but we want a space between them? Then you need to do double wrapping of string literals, etc. I think the code I wrote here is more readable. Less efficient, but it's not like this is going to be perf-critical.

Reply:
😈 

```suggestion
'Missing key "${exception.key}"' + ' ${exception.message ?? ''}'.trimRight();
```

Reply:
![image](https://user-images.githubusercontent.com/17034/109735044-e1d32280-7b76-11eb-9df2-41c0118630cf.png)